torch/utils/data/dataset.py

```diff
@@ -135,28 +142,29 @@ class IterableDataset(Dataset):
         [3, 4, 5, 6]
     """
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[T_co]:
         raise NotImplementedError
 
-    def __add__(self, other):
+    def __add__(self, other: Dataset[T_co]):
         return ChainDataset([self, other])
 
     # No `def __len__(self)` default?
     # See NOTE [ Lack of Default `__len__` in Python Abstract Base Classes ]
 
 
-class TensorDataset(Dataset):
+class TensorDataset(Dataset[Tuple[Tensor, ...]]):
     r"""Dataset wrapping tensors.
 
     Each sample will be retrieved by indexing tensors along the first dimension.
 
     Arguments:
         *tensors (Tensor): tensors that have the same size of the first dimension.
     """
+    tensors: List[Tensor]
 
-    def __init__(self, *tensors):
+    def __init__(self, *tensors: Tensor) -> None:
         assert all(tensors[0].size(0) == tensor.size(0) for tensor in tensors)
-        self.tensors = tensors
+        self.tensors = list(tensors)
```

Comment:
This changes the type of `self.tensors`, which is a public attribute, from `tuple` to `list` and does extra work - seems undesirable. Why was this change needed?